File: msite/product-webapp/webapp/src/service/Card.service.js
import axios from "axios";

// const BASE_API_URL ="http://localhost:8080/userprofile";
const BASE_API_URL ="https://MSite.stackroute.io/userprofile";

export function saveCard(Card)
{
    return axios.post(BASE_API_URL + "/card",Card);

}
export function getCardbyUsername(username)
{
    return axios.get (BASE_API_URL + "/card/user/"+username);
}

File: msite/user-profile-service/src/main/java/com/stackroute/userprofileservice/service/CardService.java
package com.stackroute.userprofileservice.service;

import com.stackroute.userprofileservice.entity.Cards;
import com.stackroute.userprofileservice.entity.User;
import com.stackroute.userprofileservice.exception.CardAlreadyExist;
import com.stackroute.userprofileservice.repository.CardsRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;

@Service
public class CardService {
    @Autowired
    private CardsRepo cardsRepo;
    public Cards saveCard(Cards cards) throws CardAlreadyExist{
        if (cardsRepo.findById(cards.getCardno()).isPresent())
        {
            throw new CardAlreadyExist();
        }
        return cardsRepo.save(cards);
    }
    public Cards getCardById(String cardsno){
        return cardsRepo.findById(cardsno).get();
    }
    public List<Cards> getCardByUserName(String username){
        return  cardsRepo.findCardsByUsername(username);
    }
}
